phases/18-ethics-safety-alignment/29-moderation-systems-openai-perspective-llamaguard/code/main.py

```
@@ -0,0 +1,126 @@
+        return "I must refuse."
+    return f"Answering: {prompt[:40]}..."
+
+
+def run(prompt: str) -> None:
+    print(f"\n>>> user: {prompt!r}")
+    # layer 1: input moderation
+    flagged_in, in_scores = input_moderator(prompt)
+    active_in = [c for c, s in in_scores.items() if s > 0]
+    if flagged_in:
+        print(f"    [INPUT FLAGGED] categories={active_in}")
+        print(f"    response: refusal")
```

Ruff (0.15.11)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)